src/esrs/aggregate.rs

```diff
@@ -48,8 +48,9 @@ pub trait Aggregate {
 /// can be persisted when handled, and the state can be reconstructed by loading and apply events sequentially.
 ///
 /// It comes batteries-included, as you only need to implement the `event_store` getter. The basic API is:
-/// 1. execute_command
+/// 1. handle_command
 /// 2. load
+/// 3. lock_and_load
 /// The other functions are used internally, but can be overridden if needed.
 #[async_trait]
 pub trait AggregateManager: Aggregate {
```

Comment:
What about transforming this AggregateManager in something similar to the command bus of the proposal of @MatteoJoliveau, allowing to publish the produced events on the event bus after the storing of the events.
The handle_command could become something like:

- load aggregate
- handle command on the aggregate
- store events
- publish the produced events on the event bus (whatever we mean with event bus 😅)

I was thinking at [this](https://github.com/primait/starsky/blob/SPIKE/cqrs/cqrs/src/command_bus.rs#L19-L39) to give the idea.

Reply:
Ehi @visd0m we are doing those kind of stuff in different PRs. [This is the PR](https://github.com/primait/event_sourcing.rs/pull/150) related to the aggregate manager (and bounds in codebase) refactor. Wdyt? We can discuss this changes in the other PR maybe?

Reply:
Sure, sorry I didn't notice the other pr 🙏.